src/ochre_simulation.py

```diff
@@ -0,0 +1,117 @@
+    xml_file = None
+    schedule_file = None
+    for file in os.listdir(input_file_path):
+        if file.endswith(".xml"):
+            xml_file = os.path.join(input_file_path, file)
+        elif file.endswith(".csv"):
+            schedule_file = os.path.join(input_file_path, file)
+
+    # Check that XML and schedule files exist. Import relevant if they do.
+    if xml_file:
+        # Extract weather station name if XML file exists
+        weatherStation = extract_weather_station(xml_file)
+    else:
+        print(f"Building XML file does not exist for: bldg{building_id:07}-up{upgrade_id:02}")
+        return
+    if not schedule_file:
+        print(f"Missing schedule file in: bldg{building_id:07}-up{upgrade_id:02}")
+
+    # Save simulation output to /ochre-sims/data/ochre_simulation/state_year_version
+    output_filepath = os.path.join(os.path.dirname(os.path.dirname(os.getcwd())), f"data/ochre_simulation/{state}_{year}_{version}/bldg{building_id:07}-up{upgrade_id:02}")
+    os.makedirs(output_filepath, exist_ok=True)
+    try:        
+        if schedule_file:
+            house = Dwelling(
+                start_time=start_time,
+                time_res=time_res,
+                duration=duration,
+                hpxml_file=xml_file,
+                hpxml_schedule_file=schedule_file,
+                output_path=output_filepath,
+                weather_file=os.path.join(os.path.dirname(os.path.dirname(os.getcwd())),"data","weather","BuildStock_TMY3_FIPS", f"{weatherStation}.epw"),
+            )
+        else:
+            house = Dwelling(
+                start_time=start_time,
+                time_res=time_res,
+                duration=duration,
+                hpxml_file=xml_file,
+                output_path=output_filepath,
+                weather_file=os.path.join(os.path.dirname(os.path.dirname(os.getcwd())),"data","weather","BuildStock_TMY3_FIPS", f"{weatherStation}.epw"),
+            )
+        df = house.simulate()
+                
+    except Exception as e:
+        print(f"Simulation failed for bldg{building_id:07}-up{upgrade_id:02}: {e}")
+        # Delete failed simulation file
+        path = os.path.join(input_file_path, f"simulation_results")
+        if os.path.exists(path):
+            try:
+                shutil.rmtree(path)
+            except Exception as e:
+                print(f"Error deleting {path}: {e}")
+        else:
+            print(f"The directory does not exist: {path}")
+        
+
+
+def extract_weather_station(file_path):
+    try:
+        with open(file_path, 'r', encoding='utf-8') as file:
+            buildingXML = xmltodict.parse(file.read())
+            weather_station_name = buildingXML['HPXML']['Building']['BuildingDetails']['ClimateandRiskZones']['WeatherStation']['Name']
+    except Exception as e:
+        print(f"Error parsing XML {file_path}: {e}")
+
+    return weather_station_name
+
+
+"""# Testing functionality
+if __name__ == '__main__':
```

Comment:
Don't comment this out, this is bit is what makes this a script rather than just a library.

That said, you should use the `click` library to create a command line interface to this entire script, and ask the user to fill in the arguments to `simulate` when the script is run.